src/server/scripts/Northrend/VioletHold/boss_ichoron.cpp

```diff
@@ -328,6 +290,146 @@ class boss_ichoron : public CreatureScript
 
             Talk(SAY_SLAY);
         }
+
+        void UpdateAI(uint32 uiDiff) override
+        {
+            if (!UpdateVictim())
+                return;
+
+            if (!bIsFrenzy && HealthBelowPct(25) && !bIsExploded)
+            {
+                Talk(SAY_ENRAGE);
+                DoCast(me, SPELL_FRENZY, true);
+                bIsFrenzy = true;
+            }
+
+            if (!bIsFrenzy)
+            {
+                if (!bIsExploded)
+                {
+                    if (!me->HasAura(SPELL_PROTECTIVE_BUBBLE))
+                    {
+                        bIsExploded = true;
+                        Talk(SAY_SHATTER);
+                        DoCast(SPELL_BURST);
+                        me->RemoveAllAuras();
+                        burstTimer = 15000;
+
+                        std::list<Creature*> summonTargets;
+                        GetCreatureListWithEntryInGrid(summonTargets, me, NPC_ICHORON_SUMMON_TARGET, 200.0f);
+                        std::list<Creature*>::iterator itr = summonTargets.begin();
+                        for (uint8 i = 0; i < 10; i++)
+                        {
+                            std::advance(itr, urand(0, summonTargets.size() - 1));
+                            Position targetPos = (*itr)->GetRandomNearPosition(10.0f);
+                            itr = summonTargets.begin();
+                            TempSummon* globule = me->SummonCreature(NPC_ICHOR_GLOBULE, targetPos, TEMPSUMMON_CORPSE_DESPAWN);
+                            DoCast(globule, SPELL_WATER_GLOBULE_VISUAL);
+
+                            float minDistance = 1000;
+                            uint8 k = 0;
```

Comment:
@Naios 
K: http://vignette1.wikia.nocookie.net/men-in-black/images/c/cb/Agent_K.jpg/revision/latest?cb=20110502202919
J: http://vignette2.wikia.nocookie.net/men-in-black/images/f/f8/Agent_J.jpg/revision/latest?cb=20110503130406